src/Renci.SshNet/Common/SshDataStream.cs

```diff
@@ -126,9 +126,17 @@ public void Write(string s, Encoding encoding)
             {
                 throw new ArgumentNullException(nameof(encoding));
             }
-
+#if NETSTANDARD2_1_OR_GREATER || NET6_0_OR_GREATER
+            ReadOnlySpan<char> value = s;
+            var count = encoding.GetByteCount(value);
+            Span<byte> bytes = stackalloc byte[count];
```

Comment:
We should guard against allocating too much on the stack

```suggestion
            Span<byte> bytes = count <= 256
                ? stackalloc byte[count]
                : new byte[count]
```